### DISCUSSION TITLE
tasks and Environment variables / $ syntax is not working on windows

### DISCUSSION THREAD

**Discussion by raphaelschnaitl** [projen/projen#3739] [Q&A]
If I add the following task to my solution:

```
const hello = project.addTask('hello');
hello.env('BAR', 'world');
hello.exec('echo $FOO, $BAR!');
```

running it on windows (pwsh) results in:
```
❯ projen hello  
👾 hello | echo $FOO, $BAR!
$FOO, $BAR!
```

executing the same task on wsl creates the expected result:
```
❯ npx projen hello
👾 hello | echo $FOO, $BAR!
, world!
```

how can I create a task that works on both platforms?

**Comment by kichik** [comment]
It's not the most portable, but I used `bash -c "echo $FOO, $BAR"` to get around this issue. It requires WSL, but it does work.

https://github.com/CloudSnorkel/cdk-github-runners/blob/1e575d17bcde6513ed8ef67f1ee70a5da7b81a14/.projenrc.js#L104-L105

**Comment by mrgrain** [comment]
I think the best way would be to write the command in JS/TS/Python/etc. There's no easy way to configure projen for that, but something like `project.addTask('node hello-world.js');` should work.
